### ISSUE TITLE
openssl crl -crlnumber is in hex and -text CRL number is in decimal, need better clarity

### ISSUE THREAD

**Issue by katajanmarja** [openssl/openssl#15800]
Hello, thank you for keeping this software available and usable for so many people!

I noticed an inconsistency in the way 'openssl crl' reports CRL numbers. It's not documented in the -help.

If I say 'openssl crl -in <file> -crlnumber', I get a hexadecimal CRL number. 

Example output: 
crlNumber=01DE

But if I say 'openssl crl -in <file> -text', I get a decimal CRL number.

Excerpt of example output: 
X509v3 CRL Number:
        478

I would expect:
1) Preferably: that the different output methods report the number the same way.  I guess -text is more used here, so it might make sense to standardize towards it and make it decimal.
2) If behaviour change is not possible or desired: That openssl crl -help would warn more visibly in the description of -crlnumber that it is hex instead of decimal. I think it warrants a warning as the difference is not obvious - you can't see from a number that it's hex unless the CRL number happens to have letters in them and can be quite misleading if not. For example 11 and 0x11 are completely different numbers.

The reason I'm thinking this behaviour might be unintentional is that usually hex is indicated e.g. with 0x, for example '0x01DE'. However, if you start to change the output string I guess it might as well just be decimal and in sync with -text, and directly readable as a number for scripting (I guess -crlnumber is more usable for scripting purposes and -text for human readability).

Openssl version I observed this with: 1.1.1

**Comment by t8m** [comment]
Unfortunately changing this in any way would almost certainly break 3rd party scripts that are using crl if they somehow use the output. I do not see many other options than just documenting this somehow in the crl manual page.

**Comment by katajanmarja** [comment]
True. Though amusingly enough, many third party scripts will break by themselves too later when CRL counts go high enough, which is likely a worse time to break than during a software upgrade.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev, barring further input

**Comment by katajanmarja** [comment]
I might confirm in case someone finds this thread by itself that this is indeed independently **fixed** (tested quickly with openssl 3.0.2, haven't gone to trace when it happened), just no one's happened to come back to mention it here.

Examples of current output
-text:
            X509v3 CRL Number: 
                5489

-crlnumber:
crlNumber=0x1571

The hex output is now clearly marked as hex output with the "0x" prefix.

Thanks a lot and have a great summer if applicable for you! :)
